Bottom: a29090aca2def9eac7e3aa97c90c8452ddaf17ef
Top:    1c5ca3939fe4b03ef193bd6a48690b2ef6b1f8e9
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Wait on openldap container startup

Occasionally the openldap container would not be ready when attempting
to add the admin and proxyagent accounts. A wait is required to
ensure the service is running properly.


---
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 9c102e67..ea01537f 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -150,6 +150,13 @@
 # *****************************************************************************
 # Set the LDAP passwords
 
+- name: wait for LDAP to start
+  wait_for:
+    delay: 5
+    host: '{{ container_addr_map.ldap.ip_addr }}'
+    port: 389
+    state: started
+
 - name: set LDAP admin password
   command: 'docker exec -i {{ openldap_container_name }} /docker-entrypoint.sh set_admin_pass {{ openldap_admin_password | quote }}'
